autospec/infile_bb_parser.py

```diff
@@ -0,0 +1,234 @@
+    ":=": "{1}",
+    "=+": "{1} {0}",
+    "+=": "{0} {1}",
+    "??=": "{1}",
+    "?=": "{1}"
+}
+
+
+def scrape_version(bb_dict):
+    """
+    Get the package version from the bb filename.
+
+    Return None if filename doesn't exist, or there is no version
+    within the filename. Otherwise, return the version.
+    """
+    try:
+        return bb_dict.get('filename')[-1].split('_', 1,)[1].rsplit('.', 1)[0]
+    except:
+        return None
+
+
+def replace_bb_variable_names(bb_dict):
+    """
+    Replace bitbake variable name with values.
+
+    Bitbake files use various variable names defined by a string
+    surrounded by ${}.
+    Variables able for replacement:
+        - ${PV} with version
+        - ${ROS_SPN} with name, or actual value if defined in .bb
+    """
+    def replace_var(bb_dict, val):
+        if bb_dict.get("VERSION"):
+            val = val.replace("${PV}", bb_dict.get('VERSION'))
+
+        if "ROS_SPN" in bb_dict:
+            val = val.replace("${ROS_SPN}", bb_dict.get("ROS_SPN"))
+        elif "NAME" in bb_dict:
+            val = val.replace("${ROS_SPN}", bb_dict.get('NAME'))
+
+        return val
+
+    for k, v in bb_dict.items():
+        if isinstance(v, list):
+            for i, j in enumerate(v):
+                v[i] = replace_var(bb_dict, j)
+        elif isinstance(v, str):
+            v = replace_var(bb_dict, v)
+
+        bb_dict[k] = v
+
+
+def get_src_url(bb_dict):
+    if bb_dict.get('SRC_URI'):
+        bb_dict["URL"] = bb_dict.get("SRC_URI").split('\\')[0].split(';')[0]
+
+
+def clean_values(value):
+    # remove beginning and end quote
+    value = value.strip('"')
+
+    # remove line continuations
+    value = value.replace("\\", "")
+
+    return value
+
+
+def read_in_command(line, depth):
+    """
+    Determines if a line is a part of a command.
+    If it returns a depth of 0, then the command has been read completely
+    from the file.
+    """
+    for c in line:
+        if c == '{':
+            depth += 1
+        if c == '}' and depth > 0:
+            depth -= 1
+        if c != "\n":
+            continue
+    return depth
+
+
+def pattern_match_line(line):
+    """
+    Determine the operation being used by the line to perform the correct
+    task on the variable. Returns the match object from searching a string
+    for the correct pattern as: [key, operator, value].
+    """
+    # escape operators for regex handling
+    operators = ["\\??\\=", "\\?\\=", "\\:\\=",
+                 "\\+\\=", "\\=\\+", "\\.\\=",
+                 "\\=\\.", "\\="]
+
+    # Split line to be [Key, operator, value] if in operators list
+    oper_pattern = r"(^[A-Z]+[_\-${}\[\]A-Za-z0-9]*)\s(" + '|'.join(
+        operators) + r")\s(\".*\")"
+
+    return re.compile(oper_pattern).search(line)
+
+
+def evaluate_expr(op, cur_val, assignment):
+    if not cur_val:
+        return assignment
+    return operators_dict.get(op).format(cur_val, assignment)
+
+
+def write_to_dict(bb_dict, m):
+
```

Comment:
Docstrings are missing on several important functions. I know much of autospec doesn't follow this standard but our new functions should.